### ISSUE TITLE
Print package version in --help section

### ISSUE THREAD

**Issue by sshine** [majjoha/passphrase#5]
When running `passphrase --help`, it would be neat if the program would mention what version is mentioned in package.yaml. Printing the version number of the program is useful for debugging purposes when filing bug reports, and keeping the version number being printed in one canonical place prevents the printed version from drifting from the actual package version.

This is possible using [`Data.Version`](https://hackage.haskell.org/package/base/docs/Data-Version.html). You can see an example [here](https://github.com/sshine/lira/blob/master/src/Main.hs#L82) / [here](https://github.com/sshine/lira/blob/master/src/Lira/Version.hs).

The trick is extracting `version` from the module `Paths_passphrase` that is dynamically generated by the Stack tool.

**Comment by majjoha** [comment]
I agree that this would be a valuable addition, and thank you for the pointers on how to implement this.

I've pushed a [branch](https://github.com/majjoha/passphrase/tree/add-version-argument) that adds support for this, but I'd prefer to not pollute `Main.hs` with concerns that relate to the CLI. When reading the documentation, it is unclear to me if `optparse-applicative` supports a behavior similar to `--help` where it simply prints the version number and exits. Do you know if this is possible?

**Comment by sshine** [comment]
Cool. :)

I don't think so, since extracting the package version number is Cabal (the library) specific.
